src/extension.ts

```diff
@@ -40,6 +42,68 @@ export async function activate(context: ExtensionContext) {
 
 }
 
+function setupDecorators(context: ExtensionContext) {
+	let timeout: NodeJS.Timer | undefined = undefined;
+	const compiletimeDecorator = vscode.window.createTextEditorDecorationType({
+		borderWidth: '1px',
+		borderStyle: 'solid',
+		overviewRulerColor: 'blue',
+		overviewRulerLane: vscode.OverviewRulerLane.Right,
+		light: {
+			// this color will be used in light color themes
+			borderColor: 'darkblue'
+		},
+		dark: {
+			// this color will be used in dark color themes
+			borderColor: 'lightblue'
+		}
+	});
+
+	let activeEditor = vscode.window.activeTextEditor;
+
+	function updateDecorations() {
+		if (!activeEditor) {
+			return;
+		}
+		const regEx = /@compiletime (\s*(static|public|private)\s)*function.+/g;
+		const text = activeEditor.document.getText();
+		const compiletime: vscode.DecorationOptions[] = [];
+		let match;
+		while (match = regEx.exec(text)) {
+			const startPos = activeEditor.document.positionAt(match.index);
+			const endPos = activeEditor.document.positionAt(match.index + match[0].length);
+			const decoration = { range: new vscode.Range(startPos, endPos), hoverMessage: 'This function will be executed at compiletime.' };
+			compiletime.push(decoration);
+		}
+		activeEditor.setDecorations(compiletimeDecorator, compiletime);
+	}
+
+	function triggerUpdateDecorations() {
+		if (timeout) {
+			clearTimeout(timeout);
+			timeout = undefined;
```

Comment:
This line could be removed. Instead `timeout` should be set to `undefined` in `updateDecorations` I think.